### PR TITLE
Add allocation analyzer and markdown linter button

### PR DESCRIPTION
**Pull request by t0ny-peng**
This PR adds Clion settings used by the allocation analyzer tool

Related to internal MR !2666

---
This change is [Reviewable](https://reviewable.io/reviews/apexai/ros2_clion_style/12)

### CODE REVIEW

**Comment by jpsamper2009** [comment]
@left4taco Could you have someone in the OS team test this out?

**Comment by t0ny-peng** [comment]
@ithier Could you review this MR?

The button looks like:
image

File: tools/External Tools.xml [hunk 1]
```diff
@@ -126,4 +126,18 @@
...
+    <exec>
+      <option name="COMMAND" value="allocation_analyzer" />
+      <option name="PARAMETERS" value="$ProjectName$ $CMakeCurrentTargetName$ --gui" />
+      <option name="WORKING_DIRECTORY" value="$ProjectFileDir$" />
+    </exec>
+  </tool>
+  <tool name="Markdown linter" description="Run markdown linter on the current file" showInMainMenu="false" showInEditor="false" showInProject="false" showInSearchPopup="false" disabled="false" useConsole="true" showConsoleOnStdOut="false" showConsoleOnStdErr="false" synchronizeAfterRun="true">
+    <exec>
+      <option name="COMMAND" value="markdownlint" />
+      <option name="PARAMETERS" value="-c ../../.markdownlint.yaml $FilePath$" />
```
Comment (orecham) [inline tools/External Tools.xml:139]: Please add the `-f` flag to auto fix those issues that can be.

Comment (t0ny-peng) [reply tools/External Tools.xml:139]: @ithier I added another button to offer more versatility. Could you review?

**Review by orecham — APPROVED** [review]
@left4taco LGTM

**Pull request merged**